signing.go

```diff
@@ -0,0 +1,105 @@
+package matrixfederation
+
+import (
+	"encoding/json"
+	"fmt"
+	"golang.org/x/crypto/ed25519"
+)
+
+// SignJSON signs a JSON object returning a copy signed with the given key.
+// https://matrix.org/docs/spec/server_server/unstable.html#signing-json
+func SignJSON(entityName, keyID string, privateKey ed25519.PrivateKey, message []byte) ([]byte, error) {
+	var object map[string]*json.RawMessage
+	var signatures map[string]map[string]Base64String
+	if err := json.Unmarshal(message, &object); err != nil {
```

Comment:
Okay, I don't really care either way - if you think this form is more helpful to the caller I'm happy with that :smile: